src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -132,6 +132,37 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
                     throw new FormatException("The assembly file version is invalid: " + fileVersion, ex);
                 }
             }
+            
+            var analyzerOptionsJson = rawProject.Value("analyzerOptions") as JsonObject;
+            if (analyzerOptionsJson != null)
+            {
+                var analyzerOptions = new AnalyzerOptions();
+                
+                foreach (var key in analyzerOptionsJson.Keys)
+                {
+                    switch (key)
+                    {
+                        case "languageId":
+                            var languageId = analyzerOptionsJson.ValueAsString(key);
+                            if (languageId == null)
+                            {
+                                throw FileFormatException.Create(
+                                    "The analyzer languageId must be a string",
+                                    analyzerOptionsJson.Value(key),
+                                    project.ProjectFilePath);
+                            }
+                            analyzerOptions.LanguageId = languageId;
```

Comment:
Right now we don't have language detection -- we have a default compiler, but that doesn't indicate language.